core/src/main/scala/cats/ApplicativeError.scala

```diff
@@ -287,6 +287,26 @@ trait ApplicativeError[F[_], E] extends Applicative[F] {
   def catchOnly[T >: Null <: Throwable]: CatchOnlyPartiallyApplied[T, F, E] =
     new CatchOnlyPartiallyApplied[T, F, E](this)
 
+  /**
+   * Often E can be created from Throwable. Here we try to call pure or
+   * catch, adapt into E, and raise.
+   *
+   * Exceptions that cannot be adapted to E will be propagated
+   */
+  def catchNonFatalAs[A](adaptIfPossible: Throwable => Option[E])(a: => A): F[A] =
```

Comment:
I would probably prefer to deal with `PartialFunction` in such a case. Not only it allows to avoid an extra allocation, but also simpler to write a handler, e.g:
```
def catchNonFatalAs[A](adaptIfPossible: PartialFunction[Throwable, E])...
```

Reply:
Not mentioning that the handler would look more like a regular `catch` in that case.

Reply:
Agreed, using PFs for error handling is a common practice 👍🏻

Reply:
I would expect most times `adaptIfPossible` will be a method reference, so it would look like this:

```scala
ApplicativeError[Either[E, *], E].catchNonFatalAs(E.fromThrowable) { 
  // block
}
```

I'm open to being wrong about this, I just expect this sort of transformation won't generally written at the call sites because, if you need to do this, you probably need to do it more than once.